1-Draft/RFCNNNN-Simplify-Module-Manifests.md

```diff
@@ -0,0 +1,170 @@
+
+# Format files (.ps1xml) to be loaded when importing this module
+# FormatsToProcess = @()
+
+# Modules to import as nested modules of the module specified in RootModule/ModuleToProcess
+# NestedModules = @()
+
+# Functions to export from this module, for best performance, do not use wildcards and do not delete the entry, use an empty array if there are no functions to export.
+FunctionsToExport = @()
+
+# Cmdlets to export from this module, for best performance, do not use wildcards and do not delete the entry, use an empty array if there are no cmdlets to export.
+CmdletsToExport = @()
+
+# Variables to export from this module
+VariablesToExport = '*'
+
+# Aliases to export from this module, for best performance, do not use wildcards and do not delete the entry, use an empty array if there are no aliases to export.
+AliasesToExport = @()
+
+# DSC resources to export from this module
+# DscResourcesToExport = @()
+
+# List of all modules packaged with this module
+# ModuleList = @()
+
+# List of all files packaged with this module
+# FileList = @()
+
+# Private data to pass to the module specified in RootModule/ModuleToProcess.
+PrivateData = @{}
+
+# HelpInfo URI of this module
+# HelpInfoURI = ''
+
+# Default prefix for commands exported from this module. Override the default prefix using Import-Module -Prefix.
+# DefaultCommandPrefix = ''
+
+# Tags applied to this module. These help with module discovery in online galleries.
+# Tags = @()
+
+# A URL to the license for this module.
+# LicenseUri = ''
+
+# A URL to the main website for this project.
+# ProjectUri = ''
+
+# A URL to an icon representing this module.
+# IconUri = ''
+
+# ReleaseNotes of this module
+# ReleaseNotes = ''
+
+}
+
+```
+
+## Specification
+
+The changes required for this RFC are relatively straightforward, and as follows:
+
+1. `New-ModuleManifest` would generate a template like what is shown above, omitting any details about the `PSData` section.
+1. Module import logic would read the manifest and pull the values for `Tags`, `LicenseUri`, `ProjectUri`, `IconUri` and `ReleaseNotes` from the top-level keys. If top-level keys were not defined with these values, it would look for values in a `PSData` section for backward compatibility.
+1. If values are defined in both locations, an error would occur informing the module author that they need to remove one of the two keys.
+1. Update `Test-ModuleManifest` to validate the keys in the new locations.
+1. Update `Update-ModuleManifest` to set the keys in the existing location if it is in use, or in the new location otherwise.
+
+## Alternate Proposals and Considerations
+
+### Move `Update-ModuleManifest` from PowerShellGet to Microsoft.PowerShell.Core
+
+This is a question for consideration: why is `Update-ModuleManifest` part of PowerShellGet and not in the Microsoft.PowerShell.Core module? Shouldn't it be co-located with the other `*-ModuleManifest` cmdlets? If so, it could be moved as part of this RFC.
```

Comment:
It would be nice. Perhaps PowerShellGet 3.0 RFC is better place to discuss the propose.

Reply:
Updated this text, because I realized having it in PowerShellGet gives downlevel support. I still think they should be co-located, though, because there is too much of an update risk having them in separate modules.